dvc/api/artifacts.py

```diff
@@ -0,0 +1,132 @@
+from typing import Any, Dict, Literal, Optional
+
+from dvc.repo import Repo
+
+from .data import _OpenContextManager, _wrap_exceptions
+
+
+def open(  # noqa, pylint: disable=redefined-builtin
+    name: str,
+    version: Optional[str] = None,
+    stage: Optional[str] = None,
+    repo: Optional[str] = None,
+    mode: Literal["r", "rb"] = "r",
+    encoding: Optional[str] = None,
+    remote: Optional[str] = None,
+    config: Optional[Dict[str, Any]] = None,
+):
+    """
+    Opens an artifact in a DVC project.
+
+    Equivalent to dvc.api.open() for a named artifact instead of a file path.
+
+    Note:
+        Use dvc.api.artifacts.read() to load the complete file contents
+        in a single function call, no context manager involved.
+        Neither function utilizes disc space.
+
+    Args:
+        name (str): name of the artifact to open.
+        version (str, optional): version of the artifact to open. Defaults to
+            the latest version.
+        stage (str, optional): name of the model registry stage.
```

Comment:
I think all these options are actually mutually exclusive. Would be good to explicitly handle that early in the code.

I am a little confused about the GTO code as it just silently overrides `tags` in:

https://github.com/iterative/gto/blob/c82563d988ea927d9cd0275bb0c2f288dd73e0b6/gto/tag.py#L158C13-L163

Reply:
Yeah, it seems like they are supposed to be mutually exclusive but I wasn't sure since GTO doesn't check for it.

CC: @aguschin